### PR TITLE
DO NOT MERGE (show diff from chronologic changes)

### PR DESCRIPTION
**Pull request by pipermerriam**
just a PR to show the diff of what's changed in chronologic's work on the codebase.

### CODE REVIEW

File: contracts/Library/SchedulerLib.sol [hunk 1]
```diff
@@ -152,25 +159,26 @@ library SchedulerLib {
...
-            ERROR();
-            msg.sender.transfer(msg.value);
-            return 0x0;
-        }
+        require( newRequestAddress != 0x0 );
+        // if (newRequestAddress == 0x0) {
+        //     // Something went wrong during creation (likely a ValidationError).
+        //     // Try to return the ether that was sent.  If this fails then
+        //     // resort to throwing an exception to force reversion.
+        //     ERROR();
```
Comment (pipermerriam) [inline contracts/Library/SchedulerLib.sol]: This block of code looks like it can be removed.

File: contracts/Scheduler/BaseScheduler.sol [hunk 2]
```diff
@@ -20,9 +20,21 @@ contract BaseScheduler is SchedulerInterface {
...
+     * @param _toAddress The address destination of the transaction.
+     * @param _callData The bytecode that will be included with the transaction.
+     * @param _uintArgs [0] The callGas of the transaction.
+     * @param _uintArgs [1] The value of ether to be send with the transaction.
+     * @param _uintArgs [2] The size of the execution window of the transaction.
+     * @param _uintArgs [3] The (block or timestamp) of when the execution window starts.
+     * @param _uintArgs [4] The gasPrice which will be used to exeute this transaction.
+     * @return The address of the new TransactionRequest.
+     */
     function scheduleTxSimple(address _toAddress,
```
Comment (pipermerriam) [inline contracts/Scheduler/BaseScheduler.sol]: I think this API should be removed in favor of a single scheduling API that *always* takes **all** scheduling parameters to require everything be set explicitly.

**Comment by pipermerriam** [comment]
This is amazing to see.  Green test suite, updated codebase, you are killing it.  ❤️

File: script/index.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+    if (requestData.schedule.windowStart < await web3.eth.getBlockNumber()) {
+        throw new Error('Cannot monitor! Window is already started.')
+    }
+
+    if (requestData.schedule.windowStart >= (await web3.eth.getBlockNumber()) + maxBlocks) {
+        throw new Error('Too far away. Please change maxBlocks variable.')
+    }
+
+    if (requestData.meta.isCancelled) throw new Error('This transaction is cancelled.')
```
Comment (pipermerriam) [inline script/index.js]: Hrm, this is problematic as it should still continue to look for more requests here, it just shouldn't add *this* request to the list of tracked requests.

File: script/index.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+    let txRequest = new web3.eth.Contract(
+        TransactionRequestABI,
+        res
+    )
+    // const requestData = await txRequest.methods.requestData().call()
+    const requestData = await RequestData.from(txRequest)
+    // log(requestData.schedule.windowStart)
+
+    if (requestData.schedule.windowStart < await web3.eth.getBlockNumber()) {
```
Comment (pipermerriam) [inline script/index.js]: This should also not fail the script entirely as the script should still look for more requests after this one until it finds that the *next* request is too far in the future.

File: script/index.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+            if (rData.meta.wasCalled) {
+                cache.delete(addr)
+            }
+            if (rData.claimData.isClaimed && (await web3.eth.getBlockNumber()) <= rData.schedule.windowStart +rData.schedule.claimWindowSize) {
+                /// nothing
+                log('shouldnt hit here yet') 
+            }
+            if (rData.claimData.isClaimed && (await web3.eth.getBlockNumber()) > rData.schedule.windowStart +rData.schedule.claimWindowSize) {
+                log('executing...')
+                txR.methods.execute().send({
```
Comment (pipermerriam) [inline script/index.js]: special care should be taken to not send the `execute()` transaction if we've already sent one and it just hasn't been mined.

**Comment by pipermerriam** [comment]
Once CI is running/passing again what do you think about getting this merged?

**Comment by lsaether** [comment]
This has been merged

**Pull request closed without merging**